internal/wasm/host.go

```diff
@@ -159,6 +164,36 @@ func addFuncs(
 	return nil
 }
 
+func addMemory(m *Module, nameToMemory map[string]*Memory) error {
+	memoryCount := uint32(len(nameToMemory))
+
+	// Only one memory can be defined or imported
+	if memoryCount > 1 {
+		memoryNames := make([]string, 0, memoryCount)
+		for k := range nameToMemory {
+			memoryNames = append(memoryNames, k)
+		}
+		sort.Strings(memoryNames) // For consistent error messages
+		return fmt.Errorf("only one memory is allowed, but configured: %s", strings.Join(memoryNames, ", "))
+	}
+
+	// Find the memory name to export.
+	var name string
+	for k, v := range nameToMemory {
+		name = k
+		if v.Min > v.Max {
+			return fmt.Errorf("memory[%s] min %d pages (%s) > max %d pages (%s)", name, v.Min, PagesToUnitOfBytes(v.Min), v.Max, PagesToUnitOfBytes(v.Max))
+		}
+		m.MemorySection = v
+	}
+
+	if name != "" {
+		m.ExportSection = append(m.ExportSection, Export{Type: ExternTypeMemory, Name: name, Index: 0})
+		m.Exports[name] = &m.ExportSection[len(m.ExportSection)-1]
+	}
+	return nil
```

Comment:
```suggestion
	for k, v := range nameToMemory {
		if v.Min > v.Max {
			return fmt.Errorf("memory[%s] min %d pages (%s) > max %d pages (%s)", name, v.Min, PagesToUnitOfBytes(v.Min), v.Max, PagesToUnitOfBytes(v.Max))
		}
		m.MemorySection = v
		m.ExportSection = append(m.ExportSection, Export{Type: ExternTypeMemory, Name: k, Index: 0})
		m.Exports[k] = &m.ExportSection[len(m.ExportSection)-1]
	}
	return nil
```

Reply:
There should only be one exported memory, so I'd rather not apply this suggestion in case there are multiple memory exports.